### PR TITLE
Refactor migrate2scratch and add installation

### PR DESCRIPTION
**Pull request by scanon**
This completely refactors migrate2scratch to be a class.  This includes tests and test data.

This also includes some unrelated installation files including python setup and RPM spec files.

### CODE REVIEW

File: tests/storage/overlay-layers/layers.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"id": "994393dc58e7931862558d06e46aa2bb17487044f670f310dffe1d24e4d1eec7", "created": "2022-09-25T21:32:34.23624767Z", "compressed-diff-digest": "sha256:213ec9aee27d8be045c6a92b7eac22c9a64b44558193775a1a7f626352392b49", "compressed-size": 2806054, "diff-digest": "sha256:994393dc58e7931862558d06e46aa2bb17487044f670f310dffe1d24e4d1eec7", "diff-size": 5826560, "compression": 2, "uidset": [0], "gidset": [0, 42]}]
```
Comment (lastephey) [inline tests/storage/overlay-layers/layers.json:1]: I think you probably want to remove this file

File: etc/nersc-plug.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ mpich:
         annotation: gov.nersc.plugins.mpich
         env: ENABLE_MPICH
         copy:
-                - /global/common/shared/das/podman/etc/02-mpich.conf:/etc/ld.so.conf.d/02-mpich.conf
+                - /etc/podman_hpc/02-mpich.conf:/etc/ld.so.conf.d/02-mpich.conf
```
Comment (lastephey) [inline etc/nersc-plug.yaml:5]: Can we make it `01-mpich.conf`?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+install:
+	python3 -m setup install --root=$(DESTDIR) --prefix=/usr
+	install -d ./etc $(DOCDIR)/
+	install ./bin/fuse-overlayfs-wrap $(DESTDIR)/usr/bin/
+	mkdir -p $(DESTDIR)/etc/podman_hpc/
+	install ./etc/nersc-plug.yaml $(DESTDIR)/etc/podman_hpc/
+	install ./etc/01-gpu.conf $(DESTDIR)/etc/podman_hpc/
+	install ./etc/02-mpich.conf $(DESTDIR)/etc/podman_hpc/
+	mkdir -p $(DESTDIR)//usr/share/containers/oci/hooks.d/
+	install ./hooks.d/02-hook_tool.json $(DESTDIR)//usr/share/containers/oci/hooks.d/
```
Comment (lastephey) [inline Makefile:17]: Can we make it `01-hook_tool.json` or `hook_tool.json`?

File: podman_hpc/podman_hpc.py [hunk 4]
```diff
@@ -68,7 +64,7 @@ def get_default_containers_conf(self):
                     'seccomp_profile': 'unconfined',
                     'runtime': self.runtime
                   }
-               }
+                }
```
Comment (lastephey) [inline podman_hpc/podman_hpc.py:67]: I always like to `except Exception as e` and print e, just for easier debugging

**Comment by lastephey** [comment]
Looks good, just a few minor (optional really) comments. There are a few files and binaries in here that I'm not sure should be stored in github, so maybe just double check.

File: etc/nersc-plug.yaml (hunk 2 above)
Comment (scanon) [reply etc/nersc-plug.yaml:5]: Sure

File: tests/storage/overlay-layers/layers.json (hunk 1 above)
Comment (scanon) [reply tests/storage/overlay-layers/layers.json:1]: This is used in a test

File: Makefile (hunk 3 above)
Comment (scanon) [reply Makefile:17]: I think it needs the prefix for the hooks.d podman thing.

File: podman_hpc/podman_hpc.py (hunk 4 above)
Comment (scanon) [reply podman_hpc/podman_hpc.py:67]: This is a fallback for cases where getlogin doesn't work.  So in that case we would just get it from the env.  I could try to catch the exact error getlogin might throw.

File: etc/nersc-plug.yaml (hunk 2 above)
Comment (scanon) [reply etc/nersc-plug.yaml:5]: Sure

File: podman_hpc/podman_hpc.py (hunk 4 above)
Comment (scanon) [reply podman_hpc/podman_hpc.py:67]: Made it just catch OSError

**Pull request merged**